snuba/admin/google.py

```diff
@@ -0,0 +1,49 @@
+from urllib.parse import urlencode
+
+import structlog
+from googleapiclient.discovery import Resource
+
+logger = structlog.get_logger().bind(module=__name__)
+
+
+def get_group_id(service: Resource, group_email: str) -> str:
+    try:
+        query_params = f"groupKey.id={group_email}"
+        request = service.groups().lookup()
+        request.uri += "&" + query_params
+        response = request.execute()
+        if "error" in response:
+            logger.exception(
+                f"An HTTP error occured when fetching group id for email {group_email}.",
+                google_api_error=response["error"],
+            )
+
+        return str(response.get("name", ""))
+    except Exception as e:
+        logger.exception(e)
```

Comment:
Essentially, there are only a few ways these can fail at all. Either something is misconfigured (e.g. the group email, infra-related config) or the google api is failing. For the user, if the check fails, they will simply not see the tool that they should otherwise see. This way, the worst that could happen is that they do not have access to a tool they should have access to which is preferable over accidentally giving someone access to a role that they don't have access to.

It might be a good idea to add a caching mechanism that caches a group's members and in case of failure, the function can check the cache. The only downside to doing this that I can think of is that in the case where someone is revoked access to a group and something fails, they will have access to a role they shouldn't have access to for the duration that the function is failing, but I think this case is extremely unlikely. We can also add caching for group IDs which I think is a good idea, but I am not sure if a google group ID ever changes, so this is something that we'd have to look into.